broker/src/main/java/org/apache/rocketmq/broker/processor/SendMessageProcessor.java

```diff
@@ -311,9 +311,9 @@ public boolean rejectRequest() {
                                                                             SendMessageContext sendMessageContext,
                                                                             ChannelHandlerContext ctx,
                                                                             int queueIdInt) {
-        return putMessageResult.thenApply((r) ->
-            handlePutMessageResult(r, response, request, msgInner, responseHeader, sendMessageContext, ctx, queueIdInt)
-        );
+        return putMessageResult.thenApplyAsync((r) ->
```

Comment:
how about  change to: 

        putMessageResult.thenAcceptAsync(result -> {
            RemotingCommand remotingCommand = handlePutMessageResult(result, response, request, msgInner, responseHeader, sendMessageContext, ctx, queueIdInt);
            if (remotingCommand != null) {
                doResponse(ctx, request, remotingCommand);
            }

        }, this.brokerController.getSendMessageExecutor());

        return null;

Reply:
i have changed

Reply:
i tried to change it, but found that
`SendMessageProcessor#processRequest` 
is invoked in 
`SendMessageProcessorTest#assertPutResult`, 
that means a returned response is necessary